### ISSUE TITLE
Ignore series?

### ISSUE THREAD

**Issue by Eagle1337** [RemiRigal/Plex-Auto-Languages#57]
There's a few series that I have that aren't super consistent with their subtitle naming or have different named japanese/english tracks (even though there's just 1) confuses the hell out of the program.

**Comment by RemiRigal** [comment]
Thank you for the suggestion, this could totally be a thing yes.
How would you like to blacklist them ? By name, by their Plex id ?

**Comment by Eagle1337** [comment]
Name seems easiest but the most prone to screw ups, but the plex id requires people knowing how to find it.

**Comment by RemiRigal** [comment]
That's exactly what I was thinking.

Another approach could be to use the Plex labels to do so, Plex-Auto-Languages could automatically ignore all tv shows that have the `PAL_IGNORE` label for example. It can be updated directly from within Plex and would not require a config change for PAL (thus no restart required), which might be simpler. What do you think ?

**Comment by Eagle1337** [comment]
That'd also work.

**Comment by dphildebrandt** [comment]
I like the idea of tagging the shows in Plex! Furthermore since you can add tags on seasons, it would be great to filter on those, by tag, as well. A great feature for anthology-like shows.

For example, Star Wars Visions: Volume 1 was all Japanese audio with English subtitles. But Volume 2 is all English with one episode being Korean.

Since my config updates by 'season' instead of 'show' its convenient to set the audio/sub for Volume 1, but allowing a episode-by-episode choice on Volume 2.

**Comment by Assaro** [comment]
Solution with tags sounds great. I would like to point out two things:
1. Make the label name configurable to avoid confusion
2. Solving that problem is in most cases a better solution. Changing subtitle tags and does not take a long time and only needs to be done once. Cant have confusion when the subs are named correctly :)

**Comment by RemiRigal** [comment]
I just merged the PR #85 that implements this.

I have added a new configuration option called `ignore_tags` which is a list of Plex tags that PAL will ignore. It defaults to `["PAL_IGNORE"]` but can be configure in any way. Only one tag from this list is required to make PAL ignore a show. A log is printed when in debug mode every time a show is ignored based on its tags.

If anyone can try this new feature I would love to get some feedback ! Thanks


> I like the idea of tagging the shows in Plex! Furthermore since you can add tags on seasons, it would be great to filter on those, by tag, as well. A great feature for anthology-like shows.

@dphildebrandt How do you add tags on season ? I cannot find the settings using the Plex web UI.

**Comment by dphildebrandt** [comment]
My mistake, you can't!

I had assumed all the same options for a show were also available for seasons. Woops!

**Comment by dphildebrandt** [comment]
@RemiRigal  Hey I'm trying to test it, I dont use a configuration file, I just set environment variables. How would I do so for this? I tried

```
pal:
    <<: *network-sec
    image: remirigal/plex-auto-languages:latest
    container_name: plex-auto-languages
    secrets:
      - plex_token
    environment:
      <<: *default-tz-puid-pgid
      PLEX_URL: http://192.168.1.29:32400
      PLEX_TOKEN_FILE: /run/secrets/plex_token
      UPDATE_LEVEL: season
      IGNORE_TAGS: PAL_IGNORE
```
I restarted it and saw in the logs
`2023-08-16 09:00:41,664 [INFO] Setting value of parameter IGNORE_TAGS from environment variable`

I tagged a show under `Tags -> Genres` watched the show, and it still updated the remaining season's audio/subs.

I then moved the tag to `Tags -> Collection` watched the show, and it still updated the remaining season's audio/subs.

I then moved the tag to `Labels -> Labels` watched the show, and it correctly output the debug log and did not change the remainder of the season!

So, I'm not sure if you intended any tagged area to work, or just the labels. It makes sense (since its not a genre or collection), but first instinct when using "tags" is to go to the `Tags` portion of the UI when editing.

Maybe update docs, or dont call it "tags" but "labels" instead?

**Comment by dphildebrandt** [comment]
Also, I'm not sure if it's intended, but as an episode is playing, it keeps pumping out the message over and over every 10 seconds
<img width="1197" alt="image" src="https://github.com/RemiRigal/Plex-Auto-Languages/assets/154459/c7f122ba-c90c-4f19-be44-5a9cb7a5c2a0">

**Comment by RemiRigal** [comment]
> Maybe update docs, or dont call it "tags" but "labels" instead?

You are absolutely right, the API had me a bit confused about the naming scheme. I will change it to `labels`, make sure that you change the name of your environment variable for the next release. Thanks !

> Also, I'm not sure if it's intended, but as an episode is playing, it keeps pumping out the message over and over every 10 seconds

Plex sends updates every 10 seconds during a play session, since the progress of the play session is not monitored for ignored shows it has to check every time. I would say that it's an intended behavior considering that this message is printed only in debug mode, but any suggestion on doing things differently is more than welcome.
